gpMgmt/bin/gppylib/programs/clsRecoverSegment.py

```
@@ -124,7 +125,16 @@ def getRecoveryActionsBasedOnOptions(self, gpEnv, gpArray):
             return GpSegmentRebalanceOperation(gpEnv, gpArray, self.__options.parallelDegree, self.__options.parallelPerHost)
         else:
             instance = RecoveryTripletsFactory.instance(gpArray, self.__options.recoveryConfigFile, self.__options.newRecoverHosts)
-            segs = [GpMirrorToBuild(t.failed, t.live, t.failover, self.__options.forceFullResynchronization) for t in instance.getTriplets()]
+            recovery_triplets = instance.getTriplets()
+            dbids_with_running_basebackup = []
+            for t in recovery_triplets:
+                if t.failed.content in self.segments_with_running_basebackup:
+                    dbids_with_running_basebackup.append(t.failed.dbid)
+
+            self.logger.warning(
```

Isn't our intention to raise an exception instead of just skipping ? The PR descriptions also mentions the same thing
> gprecoverseg, gpmovemirrors, gpaddmirrors and gpexpand should throw error if there is already a running instance of pg_basebackup for the segment that needs to be recovered/moved/added.

---

For gprecoverseg we had discussion and decided to just warn the user about segments with running basebackup and continue with recovery of others